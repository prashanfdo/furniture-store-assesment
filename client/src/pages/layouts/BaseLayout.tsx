import CategorySidebar from "components/organisms/CategorySidebar";
import Footer from "components/organisms/Footer";
import HeaderBar from "components/organisms/HeaderBar";
import React from "react";
import { useMatches } from "react-router-dom";

type BaseLayoutProps = {
  page: React.FC;
};

const BaseLayout: React.FC<BaseLayoutProps> = ({ page: Page }) => {
  const matches = useMatches();

  const categoryId = matches?.[0]?.params?.category || "156126";

  return (
    <div className="min-h-screen bg-[#eeeeef] grid grid-rows-[auto_1fr_auto]">
      <HeaderBar />
      <div className="lg:ml-[220px] mt-16">
        <CategorySidebar selectedCategoryId={categoryId} />
        <Page />
      </div>
      <Footer />
    </div>
  );
};

export default BaseLayout;
