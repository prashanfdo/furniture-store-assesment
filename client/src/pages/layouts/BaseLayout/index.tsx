import CategorySidebar from "components/organisms/CategorySidebar";
import Footer from "components/organisms/Footer";
import HeaderBar from "components/organisms/HeaderBar";
import { ROOT_CATEGORY_ID } from "helpers/constants";
import React from "react";
import { useMatches } from "react-router-dom";

type BaseLayoutProps = {
  page: React.FC;
};
const BaseLayout: React.FC<BaseLayoutProps> = ({ page: Page }) => {
  const matches = useMatches();
  const categoryId = matches?.[0]?.params?.category || ROOT_CATEGORY_ID;

  return (
    <div className="min-h-screen bg-[#eeeeef] grid grid-rows-[auto_1fr_auto]">
      <HeaderBar />
      <div className="lg:ml-[220px] mt-16">
        <CategorySidebar selectedCategoryId={categoryId} />
        <Page />
        <Footer />
      </div>
    </div>
  );
};

export default BaseLayout;
