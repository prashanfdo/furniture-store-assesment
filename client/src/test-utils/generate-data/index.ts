export * from "./article";
export * from "./category";
export * from "./search";
