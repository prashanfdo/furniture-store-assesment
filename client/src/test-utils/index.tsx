import { render, RenderOptions, renderHook, RenderHookOptions, RenderHookResult } from "@testing-library/react";
import fs from "fs";
import React, { FC, ReactElement } from "react";

import user from "@testing-library/user-event";
import { ProvidersForTests } from "providers";

type GraphQlMocks = React.ComponentProps<typeof ProvidersForTests>["mocks"];

type WrapperProps = React.ComponentProps<typeof ProvidersForTests> & {
  children: React.ReactNode;
};
const Wrapper: FC<WrapperProps> = ({ children, mocks, initialRoute }: WrapperProps) => {
  return (
    <ProvidersForTests mocks={mocks} initialRoute={initialRoute}>
      {children}
    </ProvidersForTests>
  );
};
const wrapper =
  (mocks: GraphQlMocks, initialRoute?: string) =>
  ({ children }: { children: React.ReactNode }) =>
    (
      <Wrapper mocks={mocks} initialRoute={initialRoute}>
        {children}
      </Wrapper>
    );

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper"> & { gqlMocks: GraphQlMocks; initialRoute: string },
) => {
  const view = render(ui, { wrapper: wrapper(options?.gqlMocks, options?.initialRoute), ...options });
  const style = document.createElement("style");
  style.innerHTML = fs.readFileSync("src/index.out.css", "utf8");
  document.head.appendChild(style);

  return view;
};

export const customRenderHook = <T, R>(
  callback: (props: T) => R,
  options?: Omit<RenderHookOptions<T>, "wrapper"> & { gqlMocks?: GraphQlMocks; initialRoute?: string },
): RenderHookResult<R, T> => {
  return renderHook(callback, { ...options, wrapper: wrapper(options?.gqlMocks, options?.initialRoute) });
};

export { faker } from "@faker-js/faker";
export * from "@testing-library/react";
export * from "./generate-data";
export { customRender as render };
export { customRenderHook as renderHook };
export { user };
