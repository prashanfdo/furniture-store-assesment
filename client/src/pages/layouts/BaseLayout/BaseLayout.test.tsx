import { render, screen } from "test-utils";
import BaseLayout from "./index";

jest.mock("components/organisms/CategorySidebar", () => () => <div data-testid="category-sidebar" />);
jest.mock("components/organisms/Footer", () => () => <div data-testid="footer" />);
jest.mock("components/organisms/HeaderBar", () => () => <div data-testid="header-bar" />);
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useMatches: () => jest.fn(),
}));
const dataPage = () => <div data-testid="page" />;

describe("BaseLayout", () => {
  it("should render", () => {
    const { container } = render(<BaseLayout page={dataPage} />);
    expect(container).toMatchSnapshot();
    expect(screen.getByTestId("category-sidebar")).toBeInTheDocument();
    expect(screen.getByTestId("footer")).toBeInTheDocument();
    expect(screen.getByTestId("header-bar")).toBeInTheDocument();
    expect(screen.getByTestId("page")).toBeInTheDocument();
  });
});
