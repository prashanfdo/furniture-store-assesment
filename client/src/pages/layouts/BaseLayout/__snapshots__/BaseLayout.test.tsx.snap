// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseLayout should render 1`] = `
<div>
  <div
    class="min-h-screen bg-[#eeeeef] grid grid-rows-[auto_1fr_auto]"
  >
    <div
      data-testid="header-bar"
    />
    <div
      class="lg:ml-[220px] mt-16"
    >
      <div
        data-testid="category-sidebar"
      />
      <div
        data-testid="page"
      />
      <div
        data-testid="footer"
      />
    </div>
  </div>
</div>
`;
